pkg/controllers/machine_controller.go

```diff
@@ -431,6 +431,8 @@ func (r *MachineReconciler) domainFor(
 		return nil, nil, nil, err
 	}
 
+	domainSettings.Type = "kvm"
```

Comment:
Could you please explain the context of explicitly setting this here? Also, I believe, this is not in the scope of this PR.

Reply:
Please look at my first comment for this.

Reply:
The current code implementation is already setting the correct domain type. Below is a snippet from the domainXML generated by current controller FYR.
```
<domain type='kvm' id='163'>
```
Please explain what is the need of explicitly setting this here?

Reply:
And what type of CPU is there with `kvm` domain type?

Reply:
```
<cpu mode='host-passthrough' check='none' migratable='on'/>
```

Reply:
In our test environment we are getting domain type as `qemu` which is not supporting `host-passthrough` CPU. So If we want to support only `host-passthrough` CPU, we should limit domain type also to `kvm` or remove CPU limitation.

Reply:
As I shared the snippets, the main logic is working fine.
I won't encourage to do changes in main logic to satisfy the test cases. I would suggest doing changes in test framework to get kvm as type, if that is what needed for test case.

Reply:
i agree with @so-sahu, we cannot change type of domain for test. It isn't compatible with our machines which currently exist. If you need different setup for tests, it has to be configurable and we need keep previous configuration as default.

Reply:
Removed the workaround, as it's no longer necessary. Apologies for the lengthy discussion and thanks for your patience!